apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
Funnily I had the same need.

So here is what I did:
https://github.com/mozilla-b2g/gaia/pull/30630/files#diff-c565d87c361e6319685488ffcca9557cR156
https://github.com/mozilla-b2g/gaia/pull/30630/files#diff-36ce5a6a524021e0b6c82d73d95a2a78L109
https://github.com/mozilla-b2g/gaia/pull/30630/files#diff-e93ef0174d10857928a026973f5cad18R169

I think what I did is cleaner because it handles all cases. I'd accept to rename the event to make them clearer though :)